src/main/scala/esmeta/analyzer/AbsSemantics.scala

```diff
@@ -161,6 +161,41 @@ class AbsSemantics(
   def errors: Set[TypeError] =
     errorMap.map((_, innerMap) => innerMap.values.reduce(_ + _)).toSet
 
+  def unreachables: Set[Unreachable] = {
+    val funcs = (for {
+      func <- cfg.funcs
+      entry = func.entry
+    } yield func -> entry).toSet
+    var set = Set[Unreachable]()
+    val visited = MMap[Node, Boolean]()
+    val total = npMap.keys.map(_.node).toSet
+    def checkUnreachable(node: Node) = {
+      val states = npMap
+        .filter(_._1.node == node)
```

Comment:
Probably this could be huge because afaik it iterates the whole npMap at linear time. It seems that caused by multiple Views are pointing same node, so pre-processing nodepoints without view might help since it guarantees we could search for a single element and it is much efficient in general. However, I'm not sure that retrieving a single element from a set in O(1) or O(log n) time is easy without overriding hashCode :(

Reply:
At least I think it can be achieved easily since Node elements are concretely same(even an address)- and more simple when we do not have to check for the value

Reply:
Thanks! I addressed this by extracting the node points variable, and I believe that the performance of this fix is reasonable.